examples/random_smart_tables/random_smart_tables.cc

```diff
@@ -0,0 +1,269 @@
+#include <gcs/constraints/comparison.hh>
+#include <gcs/constraints/equals.hh>
+#include <gcs/constraints/smart_table.hh>
+#include <gcs/extensional.hh>
+#include <gcs/problem.hh>
+#include <gcs/smart_entry.hh>
+#include <gcs/solve.hh>
+#include <iostream>
+#include <numeric>
+#include <random>
+#include <sstream>
+#include <string>
+#include <tuple>
+#include <vector>
+
+using namespace gcs;
+
+using std::cout;
+using std::endl;
+using std::iota;
+using std::make_optional;
+using std::nullopt;
+using std::pair;
+using std::shuffle;
+using std::string;
+using std::stringstream;
+using std::vector;
+
+using namespace innards;
+
+auto index_of(const IntegerVariableID & val, const vector<IntegerVariableID> & vec) -> int
+{
+    ptrdiff_t pos = distance(vec.begin(), find(vec.begin(), vec.end(), val));
+    return (int)pos;
+}
+
+auto random_tree_edges(int k, std::mt19937 & rng, int offset)
+{
+    vector<pair<int, int>> edges;
+
+    // Generate a random tree using Prüfer sequence
+    std::uniform_int_distribution<> rand0_to_km1(0, k - 1);
+    vector<int> prufer_seq;
+    vector<int> count_in_prufer(k, 0);
+
+    for (auto i = 0; i < k - 2; i++) {
+        auto rand_val = rand0_to_km1(rng);
+        prufer_seq.emplace_back(rand_val);
+        count_in_prufer[rand_val]++;
+    }
+
+    for (const auto & v1 : prufer_seq) {
+        for (int v2 = 0; v2 < k; v2++) {
+            if (count_in_prufer[v2] == 0) {
+                count_in_prufer[v2]--;
+                count_in_prufer[v1]--;
+                edges.emplace_back(v2 + offset, v1 + offset);
+                break;
+            }
+        }
+    }
+    auto v1 = -1;
+    int v2;
```

Comment:
Initialise because the compiler isn't smart enough to realise it will always be set at some point?